packages/next/client/index.tsx

```diff
@@ -721,7 +721,7 @@ if (process.env.__NEXT_RSC) {
   }
 
   function useServerResponse(cacheKey: string, serialized?: string) {
-    const id = (React as any).useId()
+    const id = (React as any).useId() + cacheKey
```

Comment:
Why adding the `cacheKey` here?

Reply:
(In the same file we are using `if (serverDataBuffer.has(cacheKey + ',' + id)) {` as the data key.)